web/plugin-manager/src/components/pluginBlock.tsx

```diff
@@ -0,0 +1,183 @@
+
+    // fetch info from plugin to get fresh data on demand
+    async function fetchPluginStatus(): Promise<AxiosResponse<string>> {
+        let result: AxiosResponse<string> = {} as AxiosResponse<string>;
+        try {
+            return (result = await axiosServices.getpluginInfo(dataMemoized.id));
+        } catch (error) {
+            sendErrorData(
+                "error",
+                `Plugins infos failed to get infos from plugin name : ${dataMemoized.name} on id: ${dataMemoized.id}`
+            );
+        }
+        return result;
+    }
+
+    // Launch or stop plugin
+    async function launchAndStopPlugins() {
+        let resultPluginInfo: AxiosResponse<string>;
+        // fetch info from plugin to get fresh data
+        resultPluginInfo = await fetchPluginStatus();
+        // Update data
+        setStatus(statusHelper(resultPluginInfo.data));
+
+        let result: AxiosResponse<number>;
+        if (!toggleStatus) {
+            // Launch plugin
+            try {
+                result = await axiosServices.manageLifePlugins(dataMemoized.id, "start");
+                // dataMemoized.status = PluginStatus.Up;
+                setStatus(!toggleStatus);
+                forcePlayStatus(true)
+                return result;
+            } catch (error) {
+                sendErrorData("error", "Start plugin failed");
+                // dataMemoized.status = PluginStatus.Error;
+                return;
+            }
+        } else {
+            // Stop plugin
+            try {
+                result = await axiosServices.manageLifePlugins(dataMemoized.id, "stop");
+                // dataMemoized.status = PluginStatus.Down;
+                forcePlayStatus(false)
+                setStatus(!toggleStatus);
+                return result;
+            } catch (error) {
+                sendErrorData("error", "Stop plugin failed");
+                // dataMemoized.status = PluginStatus.Error;
+                return;
+            }
+        }
+    }
+
+    // Update plugin
+    // Not implemented atm
+    function updatePlugins() {
+        //Front end update
+        // propsRef.current.updateAvailable = !propsRef.current.updateAvailable
+
+        //TODO : Uncoment this when we have a update process
+        //#####################################################
+        // const result = axiosServices.uploadPlugins("filename");
+        // if ( pluginsInfos.status && (pluginsInfos.status <= 200 || pluginsInfos.status >= 300)){
+        //     sendErrorData("error","Plugins infos failed to launch")
+        // }
+        // propsRef.current.online = true
+        // return result;
+        //#####################################################
+        console.log("Update is Not implemented ATM");
+    }
+    // Open plugin homepage
+    function openHomepagePlugins() {
+        if (statusHelper(dataMemoized.status)) window.open(dataMemoized.home);
```

Comment:
statusHelper name doesn't really help me understand what this function is doing, except helping :-)

Reply:
![image](https://user-images.githubusercontent.com/10938862/218100125-b60e86ee-e915-4927-b309-0879f2ae9aea.png)
Seems pretty clear.

It's a helper for the front, We receive string then we transform in a bool to state if we can launch or not. If you have another proposal feel free to propose.

Reply:
The function name is not clear:
- no verb used
- it doesn't explain what the function is really doing.

Using a function name sucha as: isUp(status) or isRunning(status) would make things easier to understand.

Reply:
Done